libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SFHybridViewController.m

```diff
@@ -591,8 +600,10 @@ - (void) webView:(WKWebView *) webView decidePolicyForNavigationAction:(WKNaviga
             [self refreshCredentials:[SFUserAccountManager sharedInstance].currentUser.credentials
              completion:^(SFOAuthInfo *authInfo, SFUserAccount *userAccount) {
                  [SFUserAccountManager sharedInstance].currentUser = userAccount;
+                 //Remove the session URL from the return URL as we have taken over the refresh process
+                 NSString *sessionRemovedRefreshUrl = [refreshUrl stringByReplacingOccurrencesOfString:@"%2Fvisualforce%2Fsession%3Furl%3D" withString:@""];
```

Comment:
Where is this `visualforce` URL coming from?

Reply:
No idea - it is sometimes (mostly) there and others not (when not the refresh worked before this pull request's change) - so is causes the issue.